src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -103,8 +104,78 @@ private static bool IsCodeActionNotSupportedByLSP(IUnifiedSuggestedAction sugges
         private static LSP.CodeAction[] GenerateCodeActions(
             CodeActionParams request,
             IUnifiedSuggestedAction suggestedAction,
+            LSP.CodeActionKind codeActionKind)
+        {
+            var codeAction = suggestedAction.OriginalCodeAction;
+            var diagnosticsForFix = GetApplicableDiagnostics(request.Context, suggestedAction);
+
+            using var _ = ArrayBuilder<LSP.CodeAction>.GetInstance(out var builder);
+
+            var nestedCodeActions = CollectNestedActions(request, codeActionKind, diagnosticsForFix, suggestedAction, isMainCodeAction: true);
+
+            Command? nestedCodeActionCommand = null;
+            RoslynNestedCodeAction? nestedCodeAction = null;
+            if (nestedCodeActions.Any())
+            {
+                nestedCodeAction = new RoslynNestedCodeAction(nestedCodeActions)
```

Comment:
can you make this a property, instead of a constructor parmaeter.  the mix of both is confusing.